pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -779,14 +779,37 @@ func (r *RootSyncReconciler) validateRootSecret(ctx context.Context, rootSync *v
 	return validateSecretData(rootSync.Spec.Auth, secret)
 }
 
-func (r *RootSyncReconciler) upsertClusterRoleBinding(ctx context.Context, reconcilerRef types.NamespacedName) (client.ObjectKey, error) {
-	crbRef := client.ObjectKey{Name: RootSyncPermissionsName()}
+func (r *RootSyncReconciler) configureClusterRoleBinding(ctx context.Context, reconcilerRef types.NamespacedName, roleConfig v1beta1.ClusterRoleOverrideSpec) error {
+	// in order to be backwards compatible with behavior before https://github.com/GoogleContainerTools/kpt-config-sync/pull/938
+	// we delete any ClusterRoleBinding that uses the old name.
+	// This ensures smooth migrations for users upgrading past that version boundary.
+	if err := r.deleteClusterRoleBinding(ctx, fmt.Sprintf("%s:%s", core.RootReconcilerPrefix, configsync.RootSyncName)); err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrap(err, "deleting old binding")
+	}
+
+	if roleConfig.Disabled {
+		return errors.Wrap( // wrap returns nil if err is nil
+			r.deleteClusterRoleBinding(ctx, RootSyncPermissionsName(reconcilerRef.Name)),
+			"ensuring clusterrolebinding does not exist",
+		)
+	}
+
+	if roleConfig.Name == "" {
+		roleConfig.Name = "cluster-admin"
+	}
+
+	_, err := r.upsertClusterRoleBinding(ctx, reconcilerRef, roleConfig.Name)
+	return errors.Wrap(err, "upserting cluster role binding")
+}
+
+func (r *RootSyncReconciler) upsertClusterRoleBinding(ctx context.Context, reconcilerRef types.NamespacedName, clusterRole string) (client.ObjectKey, error) {
+	crbRef := client.ObjectKey{Name: RootSyncPermissionsName(reconcilerRef.Name)}
 	childCRB := &rbacv1.ClusterRoleBinding{}
 	childCRB.Name = crbRef.Name
 
 	op, err := CreateOrUpdate(ctx, r.client, childCRB, func() error {
 		childCRB.OwnerReferences = nil
-		childCRB.RoleRef = rolereference("cluster-admin", "ClusterRole")
+		childCRB.RoleRef = rolereference(clusterRole, "ClusterRole")
 		childCRB.Subjects = addSubject(childCRB.Subjects, r.serviceAccountSubject(reconcilerRef))
```

Comment:
Do we still need to support allowing multiple subjects on the CRB? It seems like we can set a single subject since the assumption is now that each CRB maps 1:1 with a RootSync. Is the idea to allow other clients to add additional subjects to the CRB?

If we leave this as-is, it seems like we might consider keeping the cleanup logic which first checks that the subjects field is empty. WDYT?

Reply:
Hm, yeah, you're right - we should probably make this assume a single subject, and use this here
```suggestion
		childCRB.Subjects = addSubject(nil, r.serviceAccountSubject(reconcilerRef))
```
Not sure if `addSubject` is used anywhere else - if not, we could probably simplify it to e.g. not take a parameter for the existing subjects.